File: Practica 3/Python/Cola.py
class Cola:
    def __init__(self, n):
        self.arreglo = [0]*n
        self.inicio = -1
        self.fin = -1
        self.n = n
    
    def insert(self, e):
        if self.isFull():
            print("La cola está llena")
        else:
            if self.isEmpty():
                self.inicio = 0
            self.fin += 1
            self.arreglo[self.fin] = e
    
    def remove(self):
        if self.isEmpty():
            print("La cola está vacía")
            return None
        else:
            elemento = self.arreglo[self.inicio]
            self.inicio += 1
            if self.inicio > self.fin:
                self.inicio = self.fin = -1
            return elemento
    
    def peek(self):
        if self.isEmpty():
            print("La cola está vacía")
            return None
        return self.arreglo[self.inicio]
    
    def isEmpty(self):
        return self.inicio == -1
    
    def isFull(self):
        return self.fin == self.n - 1
    
    def size(self):
        if self.isEmpty():
            return 0
        return self.fin - self.inicio + 1

cola = Cola(5)
cola.insert(10)
cola.insert(20)
cola.insert(30)
print("Elemento al frente:", cola.peek())

print("Elemento eliminado:", cola.remove())
print("Elemento al frente:", cola.peek())

cola.insert(40)
cola.insert(50)
cola.insert(60)

print("¿La cola está llena?:", cola.isFull())

print("Tamaño de la cola:", cola.size())

print("Elemento eliminado:", cola.remove())
print("Elemento eliminado:", cola.remove())
print("Elemento eliminado:", cola.remove())
print("Elemento eliminado:", cola.remove())

print("¿La cola está vacía?:", cola.isEmpty())

File: Practica 3/Java/Cola/Cola.java
package practica3;

public class Cola {
    private long[] arreglo;
    private int inicio;
    private int fin;
    private int n;


    public Cola(int n) {
        this.n = n;
        arreglo = new long[n];
        inicio = -1;
        fin = -1;
    }

    public void insert(long e) {
        if (isFull()) {
            System.out.println("La cola está llena");
        } else {
            if (isEmpty()) {
                inicio = 0;
            }
            fin++;
            arreglo[fin] = e;
        }
    }

    public long remove() {
        if (isEmpty()) {
            System.out.println("La cola está vacía");
            return -1;
        } else {
            long elemento = arreglo[inicio];
            inicio++;
            if (inicio > fin) {
                inicio = fin = -1;
            }
            return elemento;
        }
    }

    public long peek() {
        if (isEmpty()) {
            System.out.println("La cola está vacía");
            return -1;
        }
        return arreglo[inicio];
    }

    public boolean isEmpty() {
        return inicio == -1;
    }

    public boolean isFull() {
        return fin == n - 1;
    }

    public int size() {
        if (isEmpty()) {
            return 0;
        }
        return fin - inicio + 1;
    }
}

